### ISSUE TITLE
Use RWMutex for accessing plugins

### ISSUE THREAD

**Issue by tedyu** [kubernetes/kubernetes#76248]
**What would you like to be added**:
Currently Mutex lock is acquired in plugins.go for accessing different maps.
Some functions only read map entry:
```
func IsFitPredicateRegistered(name string) bool {
        schedulerFactoryMutex.Lock()
        defer schedulerFactoryMutex.Unlock()
        _, ok := fitPredicateMap[name]
```
We can utilize RWMutex where the above function acquires read lock.

**Why is this needed**:
For concurrent calls to IsFitPredicateRegistered / IsPriorityFunctionRegistered, read lock would lower contention.

**Comment by tedyu** [comment]
sig/scheduling

**Comment by tedyu** [comment]
functions getFitPredicateFunctions, getPriorityFunctionConfigs and GetAlgorithmProvider can also use read lock.

**Comment by Huang-Wei** [comment]
I happened to raise #76245 earlier today.

Anyone can finish resting ones.

**Comment by tedyu** [comment]
Good to know, @Huang-Wei

**Comment by Huang-Wei** [comment]
Thanks for raising this @tedyu !

My PR #76245 covered `IsPriorityFunctionRegistered` and `GetAlgorithmProvider`. Feel free to raise another PR to cover others.

**Comment by tedyu** [comment]
Once #76245 is merged, I will send my PR

thanks

**Comment by Huang-Wei** [comment]
@tedyu it's merged now.
